core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -55,77 +56,132 @@ object ConsumerGroupCommand {
 
     val consumerGroupService = {
       if (opts.useOldConsumer) {
+        println("Note: This will only show information about consumers that use ZooKeeper (not those using the Java consumer API).\n")
         new ZkConsumerGroupService(opts)
       } else {
+        println("Note: This will only show information about consumers that use the Java consumer API (non-ZooKeeper-based consumers).\n")
         new KafkaConsumerGroupService(opts)
       }
     }
 
     try {
       if (opts.options.has(opts.listOpt))
-        consumerGroupService.list()
-      else if (opts.options.has(opts.describeOpt))
-        consumerGroupService.describe()
+        consumerGroupService.listGroups().foreach { printMessage(_) }
+      else if (opts.options.has(opts.describeOpt)) {
+        val describeGroupResult = consumerGroupService.describeGroup()
+        describeGroupResult._2 match {
+          case None =>
+            printError(s"The consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' does not exist.")
+          case Some(assignments) =>
+            if (assignments.isEmpty)
+              if (describeGroupResult._1 == "Empty")
+                printError(s"Consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' has no active member.")
+              else
+                printError(s"Consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' is rebalancing.")
+            else
+              printAssignment(assignments)
+        }
+      }
       else if (opts.options.has(opts.deleteOpt)) {
         consumerGroupService match {
-          case service: ZkConsumerGroupService => service.delete()
-          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService")
+          case service: ZkConsumerGroupService => service.deleteGroups()
+          case _ => throw new IllegalStateException(s"delete is not supported for $consumerGroupService.")
         }
       }
     } catch {
       case e: Throwable =>
-        println("Error while executing consumer group command " + e.getMessage)
-        println(Utils.stackTrace(e))
+        printError("Executing consumer group command failed due to " + e.getMessage, Some(e))
     } finally {
       consumerGroupService.close()
     }
   }
 
+  def printError(msg: String, e: Option[Throwable] = None): Unit = {
+    println("Error: " + msg)
+    e.map(debug("Exception in consumer group command", _))
+  }
+
+  def printMessage(msg: String): Unit = println(msg)
```

Comment:
Seems like this method isn't giving us much anymore. Maybe it's ok to use `println` directly?